feat(amp-deprecation): polyfills for some AMP tags

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

If a site relied on AMP, it may have some AMP tags in content. When the AMP plugin is deactivated, these tags won't render anymore, since AMP is not HTML. 

This PR adds a few light polyfills for most commonly used (and easy-to-replace) AMP tags. These polyfills are not 1:1 replacements for AMP tags – AMP-specific attributes like `layout` will not be handled. This is because AMP tags were never meant to be inserted on pages (AMP plugin should convert HTML into AMP-HTML), but if that happened for some reason, Newspack should ensure at least something is rendered when it's possible.   

### How to test the changes in this Pull Request:

1. On `master`,
1. Ensure AMP plugin is deactivated on the site
2. With a HTML block (or code editor) insert the following content into a page or post:

<details>
<summary>Click to expand code</summary>
<code>
&lt;!-- wp:html -->
&lt;h1>amp-youtube&lt;/h1>
&lt;amp-youtube data-videoid="mGENRKrdoGY" layout="responsive" width="480" height="270">&lt;/amp-youtube>
&lt;h1>amp-fit-text&lt;/h1>
&lt;amp-fit-text layout="fixed-height" min-font-size="6" max-font-size="72" height="80">&lt;h3 id="h-subscribe">Subscribe:&lt;/h3>&lt;/amp-fit-text>
&lt;h1>amp-image&lt;/h1>
&lt;amp-img width="428" height="642" layout="fixed" src="https://images.unsplash.com/photo-1677590396087-d0835963aeca?fit=crop&amp;w=428&amp;q=60" placeholder="" style="margin: auto">&lt;/amp-img>
&lt;h1>amp-iframe&lt;/h1>
&lt;amp-iframe width="200" height="100" sandbox="allow-scripts allow-same-origin" layout="responsive" frameborder="0" src="https://www.google.com/maps/embed/v1/place?key=AIzaSyAyAS599A2GGPKTmtNr9CptD61LE4gN6oQ&amp;q=iceland">
&lt;/amp-iframe>
&lt;!-- /wp:html -->
</code>
</details>

5. Visit the post, you should see nothing rendered for all AMP tags besides the `amp-fit-text`
6. Switch to this branch, reload the page, observe that the tags render a YT video, an image, and an iframe, preserving the original attributes like `width` and `height` 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

leogermani commented on includes/polyfills/class-polyfills.php:
```diff
@@ -0,0 +1,110 @@
+/**
+ * Manages settings for Polyfills.
+ */
+class Polyfills {
```
WDYT calling this Amp_Polyfills?

adekbadek replied:
Good idea, done in 59706c845288c3aca1518871289741c8bd6a59b0

leogermani reviewed (changes requested):
Looking at the result in the browser it works, but the markup has some issues.

I really think tests would help a lot here. I know you're doing a lot so I've bootstrapped a test class with some basic (failing) tests. I hope it helps. Let me know - and feel free to discard...

It has surfaced, for example, that the `img` replace leaves a `</amp-img>` closing tag behind.

And that the replacements that use the `DomDocument` insert an additional opening `<html></body>` to the document.

(fun fact: with the amp plugin enabled, I was not seeing the image nor the iframe in my tests)

adekbadek commented:
Thank you for the test setup! I've fixed the code and tweaked the tests a bit.

> with the amp plugin enabled, I was not seeing the image nor the iframe in my test

My bad! I've updated the code snippet.
